OutOfSchool/OutOfSchool.DataAccess/Models/Facebook.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+using OutOfSchool.Common;
+
+namespace OutOfSchool.Services.Models;
+
+public class Facebook
```

Comment:
We don't store them directly in the database tables, they are just entities that we use to serialize/deserialize the data. I can add comments on this. Also, I'm not sure if it would be better to use a single entity as it would add code complexity and unnecessary enum checks throughout the code.

Reply:
I actually misunderstood the question, sorry. If we use one class for all of this, we'll have to move field validation into the code, which can add complexity and lead to new bugs in the future

Reply:
I was referring to validating field attributes, but they are all the same so it doesn't matter. Hovewer I still don't think moving the OptionalContact's contacts into one class with an enum is a good idea as it will surely add complexity to the code, for example if we need to get the contact's Instagram it would look like this: contact.OptionalContact.Where(x => x.Type == ContactType.Instagram).ToList(). I can create something like OptionalContactEntityBase which will be base class for Instagram, Facebook and Website and will look like this:
``` public class OptionalContactEntityBase
{
    [DataType(DataType.Url)]
    [MaxLength(Constants.MaxUnifiedUrlLength)]
    public string Url { get; set; }
}```
to reduce repeating properties but not sure if it is needed.
I'm not that experienced in code design, so if you insist on changing the OptionalContact contacts into a single object, I'll do it.

Reply:
I'll rephrase my question: 
1.while we're gonna keep these contacts in single string it doesn't really matter how many types are there, but imagine you'll have to store them separately, how it will look like in the database ?
2. Imagine more complex case than filter just instagram contacts, for example instagram and facebook, how will you do that when all of them will be different types ?

Reply:
You're right, that looks better. I've updated the entities